### DISCUSSION TITLE
How to use Icongenie profiles for separate profiles for Capacitor iOS and Android?

### DISCUSSION THREAD

**Discussion by jjhiew** [quasarframework/quasar#14350] [CLI - Capacitor / Cordova modes]
I'm having trouble understanding how to use Icongenie profiles to generate different icon/splash sets for Android and iOS in capacitor.  I'm experiencing the same issues brought up in this comment: https://github.com/quasarframework/quasar/issues/8378.  The suggestion by @rstoenescu is to set up multiple profiles, the Quasar icongenie docs make reference to this as well:

>The most handy use case is to generate multiple profile files into one specific folder, each with their own parameters, and then run all of them through $ icongenie generate -p /path/to/folder.

So I have two profiles:

```
// icongenie-android.json
// For only generating Android icons and splash
{
  "params": {
    "skipTrim": true,
    "icon": "src/assets/android_app_icon_1024.png",
    "background": "src/assets/app_splash.png",
    "splashscreenColor": "E7F2FB",
    "splashscreenIconRatio": "0",
    "padding": [0, 0],
    "include": ["capacitor"],
    "pngColor": "E7F2FB"
  },
  "assets": [
    {
      "generator": "png",
      "folder": "src-capacitor/android/app/src/main/res/mipmap-hdpi",
      "name": "ic_launcher_foreground.png",
      "sizes": [162]
    },
   //... Only the entries for folder `src-capacitor/android`
  //... all other entries removed
  ]
}
```

```
// icongenie-ios.json
// For only generating iOS icons and splash
{
  "params": {
    "skipTrim": true,
    "icon": "src/assets/ios_app_icon_1024.png",
    "background": "src/assets/app_splash.png",
    "splashscreenColor": "E7F2FB",
    "splashscreenIconRatio": "0",
    "padding": [0, 0],
    "include": ["spa", "capacitor"],
    "pngColor": "E7F2FB"
  },
  "assets": [
    {
      "generator": "png",
      "name": "AppIcon-20x20@1x.png",
      "folder": "src-capacitor/ios/App/App/Assets.xcassets/AppIcon.appiconset",
      "sizes": [20],
      "background": true
    },
   //... Only the entries for folder `src-capacitor/ios`
   //... all other entries removed
  ]
}
```

The profile is that if I run `icongenie generate -p <profile>` using either of the two profiles above, it always generates icon sets and splashs for BOTH ios and android. 

Any help would be much appreciated on how I can generate icon sets for each platform that won't overwrite the other.

**Comment by bonatoc** [comment]
One way to circumvent this is to let icongenie generate the assets for both platforms, and then rewrite the Android assets using Android Studio. On the link below you can find the instructions for Android Studio:

[https://www.joshmorony.com/adding-icons-splash-screens-launch-images-to-capacitor-projects/](https://www.joshmorony.com/adding-icons-splash-screens-launch-images-to-capacitor-projects/)
